### ISSUE TITLE
Train accuracy 100% and test accuracy 0 percent

### ISSUE THREAD

**Issue by Z-Rajaei** [rusty1s/pytorch_geometric#2065]
## ❓ Questions & Help

Hello, I have a problem with my Pytorch Geometric code, I hope someone can help me!
I have 2000 graphs, I have converted them to pytorch geometric data using the my modified form_networkx function.  But the training and testing result is not as desired. I get 100 percent train accuracy and zero percent test accuracy  in all epochs!!
I think something is wrong with my code. (I do not think the problem is with the data!)

Can anybody help me find the problem??

    def from_networkx2(self, G):
        r"""Converts a :obj:`networkx.Graph` or :obj:`networkx.DiGraph` to a
        :class:`torch_geometric.data.Data` instance.

        Args:
            G (networkx.Graph or networkx.DiGraph): A networkx graph.
        """

        G = nx.convert_node_labels_to_integers(G)
        G = G.to_directed() if not nx.is_directed(G) else G
        edge_index = torch.tensor(list(G.edges)).t().contiguous()

        data = {}
        attrdata = {}

        for i, (_, feat_dict) in enumerate(G.nodes(data=True)):
            for key, value in feat_dict.items():
                attrdata[str(key)] = [value] if i == 0 else attrdata[str(key)] + [value]

        for i, (_, _, feat_dict) in enumerate(G.edges(data=True)):
            for key, value in feat_dict.items():
                attrdata[str(key)] = [value] if i == 0 else attrdata[str(key)] + [value]

        for key, item in attrdata.items():
            try:
                attrdata[key] = torch.tensor(item)
            except ValueError:
                pass

        # Converting features to one_hot encoding, concatinating them and creating Data.x
        for key, item in attrdata.items():
            try:
                attrdata[key] = torch.nn.functional.one_hot(attrdata[key],num_classes=100).to(torch.float)

            except ValueError:
                pass
        a=[]
        i=0
        for key, item in attrdata.items():
            try:
                a.append(attrdata[key])

            except ValueError:
                pass

        data['x'] = torch.cat( a, dim=-1)
        # up to here

        data['edge_index'] = edge_index.view(2, -1)
        data = torch_geometric.data.Data.from_dict(data)
        data.num_nodes = G.number_of_nodes()
        return data

    def process(self):

        data_list = []
        for graph in self.Graph:
            data_list.append(self.from_networkx2(graph))

        data, slices = self.collate(data_list)

        torch.save((data, slices), self.processed_paths[0])

These are some of my codes that me be problematic!
And, my training code is as bellow:

```
class Net(torch.nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        self.conv1 = GCNConv(dataset.num_node_features, 4)
        self.conv2 = GCNConv(4, 4)
        self.lin = torch.nn.Linear(4, dataset.num_classes)

    def forward(self, data):
        x, edge_index, batch = data.x, data.edge_index, data.batch

        x = self.conv1(x, edge_index)
        x = F.relu(x)
        x = F.dropout(x, training=self.training)
        x = self.conv2(x, edge_index)
        x = F.relu(x)
        x = global_mean_pool(x, batch)
        x = self.lin(x)

        return F.log_softmax(x, dim=1)


def train(epoch):
    model.train()

    loss_all = 0
    for data in train_loader:
        data = data.to(device)
        optimizer.zero_grad()
        output = model(data)
        loss = F.nll_loss(output, data.y)
        loss.backward()
        loss_all += data.num_graphs * loss.item()
        optimizer.step()
    return loss_all / len(train_dataset)


def test(loader):
    model.eval()

    correct = 0
    for data in loader:
        data = data.to(device)
        pred = model(data).max(dim=1)[1]
        correct += pred.eq(data.y).sum().item()
    return correct / len(loader.dataset)


n = len(dataset) // 2
test_dataset = dataset[:n]
train_dataset = dataset[n:]
test_loader = DataLoader(test_dataset, batch_size=80)
train_loader = DataLoader(train_dataset, batch_size=80)

device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
model = Net().to(device)

optimizer = torch.optim.Adam(model.parameters(), lr=0.01, weight_decay=5e-4)

for epoch in range(1, 30):
    loss = train(epoch)
    train_acc = test(train_loader)
    test_acc = test(test_loader)
    print('Epoch: {:03d}, Loss: {:.5f}, Train Acc: {:.5f}, Test Acc: {:.5f}'.
          format(epoch, loss, train_acc, test_acc))

```
The first epochs are as folows:
Epoch: 001, Loss: 0.97482, Train Acc: 1.00000, Test Acc: 0.00000
Epoch: 002, Loss: 0.46829, Train Acc: 1.00000, Test Acc: 0.00000
Epoch: 003, Loss: 0.06011, Train Acc: 1.00000, Test Acc: 0.00000
Epoch: 004, Loss: 0.01299, Train Acc: 1.00000, Test Acc: 0.00000
Epoch: 005, Loss: 0.00604, Train Acc: 1.00000, Test Acc: 0.00000
Epoch: 006, Loss: 0.00346, Train Acc: 1.00000, Test Acc: 0.00000

Any help from anybody is appreciated!!!

**Comment by rusty1s** [comment]
Did you resolve it?

**Comment by Z-Rajaei** [comment]
Yes, I made some changes. Now the output is very good now.
In some runs the loss value in the first epochs is greater than 1 (such as 1.04235 and 1.01808). Are these loss values valid?

**Comment by rusty1s** [comment]
The loss value can be greater than 1, this is not a problem at all.

**Comment by Z-Rajaei** [comment]
Great! Thank you very much.

**Comment by CemBirbiri** [comment]
I have a problem with converting the networkx Graphs to torch_geometric Data. 
Which dataset did you use here?
